src/Authentication.Abstractions/Interfaces/IConfigManager.cs

```diff
@@ -0,0 +1,92 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.PowerShell.Common.Config
+{
+    public interface IConfigManager
+    {
+        /// <summary>
+        /// Path of the config file.
+        /// </summary>
+        string ConfigFilePath { get; }
+
+        /// <summary>
+        /// Register a config.
+        /// </summary>
+        /// <remarks>Register all the configs before <see cref="BuildConfig"/></remarks>
+        void RegisterConfig(ConfigDefinition config);
+
+        /// <summary>
+        /// Retrieve data from all the providers and build config values.
+        /// </summary>
+        void BuildConfig();
+
+        /// <summary>
+        /// Get the value of a config by key.
+        /// </summary>
+        /// <typeparam name="T">Type of the value.</typeparam>
+        /// <param name="key">Key of the config.</param>
+        /// <param name="invocation">PowerShell cmdlet invocation info. If not null, the config that matches the module or cmdlet name will be returned.</param>
+        /// <remarks>For the list of available keys, see <see cref="ConfigKeys"/>, for those used in service projects, or see <see cref="ConfigKeysForCommon"/> for those used in common projects.
+        /// The `invocation` parameter is typed `object` because we don't want Authentication.Abstractions project to take dependency on PowerShell SDK.
+        /// However at runtime it needs to be of type `InvocationInfo`.</remarks>
+        /// <returns>Value of the config, or the default value if never set.</returns>
+        T GetConfigValue<T>(string key, object invocation = null);
```

Comment:
`invocation` was made an `object` type as we don't want to take dependency on the PowerShell SDK in the `Authentication.Abstractions` project.